Batteries/Data/Vector/Unexpanders.lean

```diff
@@ -0,0 +1,138 @@
+/-
+Copyright (c) 2024 Shreyas Srinivas. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Shreyas Srinivas, Francois Dorais
+-/
+
+import Batteries.Data.Vector.Basic
+
+/-!
+## Vector.Unexpanders
+
+The definitions in this file are used to provide unexpanders in metaprogramming.
+They correspond to similar array definitions `mkArray0`,...,`mkArray8` and their unexpanders
+-/
+
+namespace Vector
+
+/-- Create empty vector `#v[]`-/
+def mkVector0 : Vector α 0 := empty
```

Comment:
I don't think we need these functions. The corresponding functions in core have a specialized purpose, and I think we should add these only if we need them and have a specific application for them.

Reply:
Wasn't there an interest in using this interface eventually in core?

Reply:
The description also doesn't match the one I suggested but that's lost now.

Reply:
> The description also doesn't match the one I suggested but that's lost now.

Are you referring to the module doc?

Reply:
Yes.

Reply:
If the consensus is that `Unexpander.lean` should be removed then it can be done. But what is their purpose. Does it affect how vectors will be unexpanded in any way during proofs?

Reply:
I found the module doc.

Reply:
That's the one.

Reply:
A bit moot now since this file is to be deleted.

Reply:
@digama0 can I make a separate PR for deleting this file. This is to maintain the PR dependency requirement you suggested.

Reply:
Not a dependency anymore. Unexpander can be deleted.

Reply:
Done.